src/components/onCall/oncall-list-view.tsx

```diff
@@ -38,35 +39,23 @@ const OnCallListView = ({ selectedDate, setSelectedDate, updatePaidStatus }: Pro
     return loggedInEmail && email && loggedInEmail.toLowerCase() === email.toLowerCase();
   };
 
+  /**
+   * Handle checkbox change for paid status
+   * Make API call to update status and update local state on success
+   *
+   * @param week - Week number
+   * @param currentPaid - Current paid status
+   *
+   */
+  const handleCheckboxChange = async (week: number, currentPaid: boolean) => {
+    try {
+      await updatePaidStatus(selectedDate.year, week, currentPaid);
```

Comment:
If the checkbox value is rendered straight from the data from onCallData then will the atom update and the compoent re render based on this request?
If not it might be worth handling with some state as well. And even if it does should we still have some kind of loading indication on the checkbox to indicate there is a request/ something is happening?

This depends on the UI- your call what is best fit here